classes/class-gf-securesubmit.php

```diff
@@ -1342,23 +1346,22 @@ protected function buildCheckHolder($feed, $submission_data, $entry) {
      *
      * @return \HpsAddress
      */
-    private function buildAddress($feed, $entry) {
+    private function buildAddress($feed, $entry)
+    {
         $address = new HpsAddress();
-        if (in_array('billingInformation_address', $feed['meta'])) {
-            $address->address
-                = $entry[ $feed['meta']['billingInformation_address'] ] . $entry[ $feed['meta']['billingInformation_address2'] ];
-        }
-        if (in_array('billingInformation_city', $feed['meta'])) {
-            $address->city = $entry[ $feed['meta']['billingInformation_city'] ];
-        }
-        if (in_array('billingInformation_state', $feed['meta'])) {
-            $address->state = $entry[ $feed['meta']['billingInformation_state'] ];
+        $meta = $this->get_address_card_field($feed);
+        //$meta->get
+        $address->address = rgar($entry, $meta->id . '.1');
+        $address->city = rgar($entry, $meta->id . '.3');
+        $address->state = rgar($entry, $meta->id . '.4');
+        $address->zip = rgar($entry, $meta->id . '.5');
+        //'United States' 'Canada'
+        $address->country = rgar($entry, $meta->id . '.6', 'USA');;
```

Comment:
It doesn't appear so, but does this change still respect mapped fields within a feed's settings?